src/main/java/net/dv8tion/jda/core/entities/EntityBuilder.java

```diff
@@ -986,6 +986,42 @@ public PermissionOverride createPermissionOverride(JSONObject override, Channel
                 .setDeny(deny);
     }
 
+    public Webhook createWebhook(JSONObject object)
+    {
+        String id = object.getString("id");
+        String token = !object.isNull("token") ? object.getString("token") : null;
+        String guildId = object.getString("guild_id");
+        String channelId = object.getString("channel_id");
+
+        Object name = !object.isNull("name") ? object.get("name") : JSONObject.NULL;
+        Object avatar = !object.isNull("avatar") ? object.get("avatar") : JSONObject.NULL;
+
+        JSONObject fakeUser = new JSONObject()
+                    .put("username", name)
+                    .put("discriminator", "0000")
+                    .put("id", id)
+                    .put("avatar", avatar);
+        User defaultUser = createFakeUser(fakeUser, false);
+
+        JSONObject ownerJson = object.getJSONObject("user");
+        String userId = ownerJson.getString("id");
+
+        User owner = api.getUserById(userId);
+        if (owner == null)
+        {
+            fakeUser.put("id", userId);
+            fakeUser.put("username", ownerJson.getString("username"));
+            fakeUser.put("avatar", !ownerJson.isNull("avatar") ? ownerJson.getString("avatar") : JSONObject.NULL);
+            fakeUser.put("discriminator", ownerJson.getString("discriminator"));
```

Comment:
Why are you crafting a new `fakeUser` json when it appears that `ownerJson` already had all of the needed information? The only thing that seems to possible be missing from `ownerJson` is `userId`, and if that is the case just `ownerJson.put("id", userId)`. I see no need to create another json body. Is there a good reason for it?